contracts/badger-ren/BadgerRenAdapter.sol

```diff
@@ -0,0 +1,272 @@
+        exchange = ICurveExchange(_exchange);
+        renBTC = registry.getTokenBySymbol("BTC");
+        wBTC = IERC20(_wbtc);
+
+        mintFeeBps = _feeConfig[0];
+        burnFeeBps = _feeConfig[1];
+        percentageFeeRewardsBps = _feeConfig[2];
+        percentageFeeGovernanceBps = _feeConfig[3];
+
+        // Approve exchange.
+        require(renBTC.approve(_exchange, uint256(-1)));
+        require(wBTC.approve(_exchange, uint256(-1)));
+    }
+
+    function recoverStuck(
+        bytes calldata encoded,
+        uint256 _amount,
+        bytes32 _nHash,
+        bytes calldata _sig
+    ) external nonReentrant {
+        // Ensure sender matches sender of original tx.
+        uint256 start = encoded.length - 32;
+        address sender = abi.decode(encoded[start:], (address));
+        require(sender == msg.sender);
+
+        bytes32 pHash = keccak256(encoded);
+        uint256 _mintAmount = registry.getGatewayBySymbol("BTC").mint(pHash, _amount, _nHash, _sig);
+        uint256 _fee = _processFee(renBTC, _mintAmount, mintFeeBps);
+
+        emit RecoverStuckRenBTC(_mintAmount, _fee);
+
+        renBTC.safeTransfer(msg.sender, _mintAmount.sub(_fee));
+    }
+
+    function mintRenBTC(
+        // user args
+        address payable _destination,
+        // darknode args
+        uint256 _amount,
+        bytes32 _nHash,
+        bytes calldata _sig
+    ) external nonReentrant {
+        // Mint renBTC tokens
+        bytes32 pHash = keccak256(abi.encode(_destination));
+        uint256 _mintAmount = registry.getGatewayBySymbol("BTC").mint(pHash, _amount, _nHash, _sig);
+        uint256 _fee = _processFee(renBTC, _mintAmount, mintFeeBps);
+
+        emit MintRenBTC(_mintAmount, _fee);
+
+        renBTC.safeTransfer(_destination, _mintAmount.sub(_fee));
+    }
+
+    function burnRenBTC(bytes calldata _btcDestination, uint256 _amount) external nonReentrant {
+        require(renBTC.balanceOf(address(msg.sender)) >= _amount);
+        uint256 _startBalance = renBTC.balanceOf(address(this));
+        renBTC.safeTransferFrom(msg.sender, address(this), _amount);
+        uint256 _endBalance = renBTC.balanceOf(address(this));
+
+        uint256 _burnAmount = _endBalance.sub(_startBalance);
+        uint256 _fee = _processFee(renBTC, _burnAmount, burnFeeBps);
+
+        emit BurnRenBTC(_burnAmount, _fee);
+
+        uint256 burnAmount = registry.getGatewayBySymbol("BTC").burn(_btcDestination, _burnAmount.sub(_fee));
+    }
+
+    function mintWBTC(
+        uint256 _slippage,
+        address payable _destination,
+        uint256 _amount,
+        bytes32 _nHash,
+        bytes calldata _sig
+    ) external nonReentrant {
+        // Mint renBTC tokens
+        bytes32 pHash = keccak256(abi.encode(_slippage, _destination));
+        uint256 _mintAmount = registry.getGatewayBySymbol("BTC").mint(pHash, _amount, _nHash, _sig);
+        _swapRenBTCForWBTC(_slippage, _destination, _mintAmount);
+    }
+
+    function _swapRenBTCForWBTC(
```

Comment:
Should author consider implementing swap token method such that it has dependency on smart contract that does the swap therefore allowing for greater modularity and interchangeability in the future? ATM swap is hardcoded to curve.fi which might not be the most optimal place to swap pegged btc around(specially for big amounts). I suspect that in rare but possible scenario where curve pool becomes unbalanced and the user insists on swap with slip limit it will render mint functionality  for wBTC unusable

1inch has off chain facility to find optimal routing for swaps, I am not certain how big of priority is supporting large swaps with limited slip?